### PR TITLE
tickets/PREOPS-5360: clean up the nightsum and prenight Times Square notebooks by updating schedview to support them better

### PR DESCRIPTION
**Pull request by ehneilsen**

### CODE REVIEW

**Comment by rhiannonlynne** [comment]
Can you add a screenshot of what gets produced by `create_metric_visit_map_grid`? 
I think I know but want to make sure - I think the model observatory and thus conditions are only used to put in the sun and moon positions right? Is this function useful enough that it should be available even if conditions are not available, and then the sun/moon are just skipped?

**Comment by rhiannonlynne** [comment]
Looks fine, but ruff failure.

**Review by rhiannonlynne — APPROVED** [review]

File: schedview/collect/opsim.py [hunk 1]
```diff
@@ -122,7 +124,13 @@ def read_opsim(
...
         with sqlite3.connect(local_obs_path.ospath) as sim_connection:
             try:
-                visits = pd.DataFrame(maf.get_sim_data(sim_connection, constraint, dbcols, **kwargs))
+                try:
+                    visits = pd.DataFrame(maf.get_sim_data(sim_connection, constraint, dbcols, **kwargs))
+                except UserWarning:
+                    warn("No visits match constraints.")
+                    visits = pd.DataFrame(rubin_scheduler.scheduler.utils.empty_observation()).drop(index=0)
+                    if "observationId" not in visits.columns and "ID" in visits.columns:
+                        visits.rename(columns={"ID": "observationId"}, inplace=True)
```
[the remark this answers is not in the document]
Comment (ehneilsen) [reply schedview/collect/opsim.py:101]: At present it doesn't matter, but you are correct that using SchemaConverter.obs2opsim will get it right, whereas what's there now will not. Fixing it now.

**Comment by ehneilsen** [comment]
`create_metric_visit_map_grid` creates the maps of most recent and accumulated visits in the nightsum pages. The model observatory is used not only for the sun and moon positions, but also to get the observatory latitude which is needed for the horizon circles at sunrise and sunset.
Could it be optional so this could be run without a model observatory, if we did not plot these things? Sure, but I suspect that someone doing this might just want to go directly to the metric plotting tools in maf, or use maf plus raw uranography of the healpix map: much of the point in these functions is to put together all the elements needed for the specific plots needed for the nightsum and similar notebooks, so I don't think making the model observatory optional will be of much use.

**Comment by ehneilsen** [comment]
Sample output from create_metric_visit_map metric_visit_map.

**Pull request merged**